src/OrchardCore/OrchardCore.Settings.Core/Deployment/SiteSettingsPropertyDeploymentSource.cs

```diff
@@ -0,0 +1,33 @@
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+using OrchardCore.Deployment;
+using OrchardCore.Entities;
+
+namespace OrchardCore.Settings.Deployment
+{
+    public class SiteSettingsPropertyDeploymentSource<TModel> : IDeploymentSource where TModel : class, new()
+    {
+        private readonly ISiteService _siteService;
+
+        public SiteSettingsPropertyDeploymentSource(ISiteService siteService)
+        {
+            _siteService = siteService;
+        }
+
+        public async Task ProcessDeploymentStepAsync(DeploymentStep step, DeploymentPlanResult result)
+        {
+            var settingsStep = step as SiteSettingsPropertyDeploymentStep<TModel>;
+            if (settingsStep == null)
+            {
+                return;
+            }
+
+            var siteSettings = await _siteService.GetSiteSettingsAsync();
+
+            result.Steps.Add(new JObject(
```

Comment:
We should find the property if it exists already, so it is only one step that is being added.
I will do this, while I am checking the registration with both attributes (i.e. `Feature` and `RequiresFeature` works ok)

Reply:
@agriffard have changed it to look for an existings `Settings` property rather than creating a new one every time.

output changes from
```
  "steps": [
    {
      "name": "Settings", <-- first Settings step
      "AdminSettings": {
        "DisplayMenuFilter": false
      }
    },
    {
      "name": "Settings", <-- second Settings step
      "LoginSettings": {
        "UseSiteTheme": false,
        "UseExternalProviderIfOnlyOneDefined": false,
        "DisableLocalLogin": false,
        "UseScriptToSyncRoles": false,
        "SyncRolesScript": null
      }
    }
  ]
```

```
  "steps": [
    {
      "name": "Settings", <-- Single Settings step
      "AdminSettings": {
        "DisplayMenuFilter": false
      },
      "LoginSettings": {
        "UseSiteTheme": false,
        "UseExternalProviderIfOnlyOneDefined": false,
        "DisableLocalLogin": false,
        "UseScriptToSyncRoles": false,
        "SyncRolesScript": null
      },
      "HttpsSettings": {
        "EnableStrictTransportSecurity": false,
        "RequireHttps": false,
        "RequireHttpsPermanent": false,
        "SslPort": null
      }
    }
  ]
```

An optimization, and makes it clearer that all the steps are contained in one 

If you are happy with the change let me know, and I think we can merge it.

I've checked the `Feature` and `RequiresFeature` are working fine together. i.e. if you disable deployment but still have registration the startup doesn't fire.

Reply:
Great